custom_components/opensprinkler/time.py

```diff
@@ -67,10 +76,24 @@ def icon(self) -> str:
     @property
     def native_value(self) -> time:
         """The value of the time."""
+        # Only start0 supported at this time.
         start_index = 0
-        minutes = self._program.program_start_times[start_index]
-        the_time = datetime.time(round(minutes / 60), minutes % 60, 0)
-        return the_time
+        start0 = self._program.program_start_times[start_index]
+        sign = -1 if is_set(start0, START_TIME_SIGN_BIT) else 1
+
+        if is_set(start0, START_TIME_SUNSET_BIT):
+            minutes = self._controller.sunset + (
+                (start0 & START_TIME_MINUTES_MASK) * sign
+            )
+        elif is_set(start0, START_TIME_SUNRISE_BIT):
+            minutes = self._controller.sunrise + (
+                (start0 & START_TIME_MINUTES_MASK) * sign
+            )
+        else:
+            minutes = start0
+
+        start_time = datetime.time(trunc(minutes / 60), minutes % 60, 0)
+        return start_time
```

Comment:
Are you planning to keep this logic here or move it to pyopensprinkler eventually? 

I was wondering if `program_start_times` should always just return the time and have properties for each bit/flag and even the offset in pyopensprinkler. What do you think?

Reply:
Sorry, I was planning to leave a comment about the bit twiddling and got busy. 

But yes, the bit logic belongs in pyopensprinkler where your other bit logic lives. I was just trying to get the issue for sunrise/sunset folks resolved quickly, and given the issues some have had with updating, wanted to keep the releases of pyopensprinkler to a minimum. I figured I would move it later, possibly when adding support for Additional Start Times. 

As to the exact implementation details for moving the logic, I hadn't thought that far ahead, except that the UI shouldn't change. I do find it useful to have the type of offset, the offset amount, and the actual time (directly settable) available to the user. For example, all three of these entities are editable:

![StartTime](https://github.com/vinteo/hass-opensprinkler/assets/56356940/36b84ffe-39f2-4ddf-a6c2-9b5facc59762)